test/controllers/authentication/source_ip_checker_test.rb

```diff
@@ -30,24 +29,31 @@ class SourceIpCheckerTest <  ActiveSupport::TestCase
   end
 
   test 'does not allow ip if unknown location' do
-    country = {}
-    MaxMind::DB.any_instance.expects(:get).with('1.42.42.12').returns(country)
-
+    GeoIp.expects(:activated?).returns(true).at_least_once
     checker = Authentication::SourceIpChecker.new('1.42.42.12')
-
+    geo_ip = mock('geo_ip');
+    checker.expects(:geo_ip).returns(geo_ip)
+    geo_ip.stubs(:country_code)
     assert_equal false, checker.ip_authorized?
   end
 
   test 'does not allow ip if country is not whitelisted' do
+    GeoIp.expects(:activated?).returns(true).at_least_once
     checker = Authentication::SourceIpChecker.new('8.8.8.8')
+    geo_ip = mock('geo_ip');
+    checker.expects(:geo_ip).returns(geo_ip).at_least_once
+    geo_ip.expects(:country_code).returns('US').at_least_once
     assert_equal false, checker.send(:country_authorized?)
     assert_equal false, checker.ip_authorized?
   end
 
   test 'allows ip from whitelisted country' do
+    GeoIp.expects(:activated?).returns(true).at_least_once
     ch_ip = '46.140.0.1'
-
     checker = Authentication::SourceIpChecker.new(ch_ip)
+    geo_ip = mock('geo_ip');
+    checker.expects(:geo_ip).returns(geo_ip).at_least_once
```

Comment:
just an idea ...
add a method -> mock_geoip(checker)
since you're using it in multiple test cases